internal/querycoordv2/task/merger.go

```diff
@@ -126,32 +104,31 @@ func (merger *Merger[K, R]) Add(task MergeableTask[K, R]) {
 	merger.waitQueue <- task
 }
 
-func (merger *Merger[K, R]) merge(id K, queue chan MergeableTask[K, R]) {
-	if merger.isStopped() {
-		return
-	}
-	if !merger.processors.Insert(id) {
-		return
+func (merger *Merger[K, R]) drain() {
```

Comment:
> this breaks the limit `Params.QueryCoordCfg.TaskMergeCap`, the number of tasks to merge, is this on purpose?

Don't see any reason why we want to keep TaskMergeCap? is it because of too many tasks will cause timeout?

Reply:
if we want to follow it, maybe just break when TaskMergeCap if any queue is larger than TaskMergeCap?

Reply:
Too many tasks may fail with higher possibility, (as only one segment of them be compacted). Timeout has been removed.